Sources/SecureXPC/Server/XPCAnonymousServer.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  XPCAnonymousServer.swift
+//  
+//
+//  Created by Alexander Momchilov on 2021-11-28.
+//
+
+import Foundation
+
+internal class XPCAnonymousServer: XPCServer {
+    private let anonymousListenerConnection: xpc_connection_t
+
+    internal override init() {
+        self.anonymousListenerConnection = xpc_connection_create(nil, nil)
+        super.init()
+
+        // Start listener for the new anonymous connection, all received events should be for incoming client connections
+         xpc_connection_set_event_handler(anonymousListenerConnection, { newClientConnection in
+             // Listen for events (messages or errors) coming from this connection
+             xpc_connection_set_event_handler(newClientConnection, { event in
+                 self.handleEvent(connection: newClientConnection, event: event)
+             })
+             xpc_connection_resume(newClientConnection)
+         })
+         xpc_connection_resume(anonymousListenerConnection)
+    }
+
+    public override func start() -> Never {
+        fatalError("Anonymous services don't need to be `start()`ed.")
```

Comment:
See above comment in regards to some of the initializer functionality being relocated to here.